### PR TITLE
refac: remove string utils

### PR DESCRIPTION
**Pull request by saiintbrisson**

### CODE REVIEW

File: pallets/subspace/src/registration.rs [hunk 1]
```diff
@@ -38,13 +36,14 @@ impl<T: Config> Pallet<T> {
 		);
 
 		// --- 4. Resolve the network in case it doesn't exist
-		if !Self::subnet_name_exists(network.clone()) {
-			// If the subnet doesn't exist, registration will create it.
-			Self::add_subnet_from_registration(network.clone(), stake_amount, &key)?;
-		}
+		let netuid = if let Some(netuid) = Self::get_netuid_for_name(&network) {
```
Comment (aripiprazole) [inline pallets/subspace/src/registration.rs:39]: .unwrap_or_else

File: pallets/subspace/src/voting.rs [hunk 2]
```diff
@@ -182,16 +186,13 @@ impl<T: Config> Pallet<T> {
 		let mode = proposal.mode.clone();
 
 		// check if proposal is valid
-		if is_vec_str(mode.clone(), "global") {
+		if mode == GLOBAL_MODE {
```
Comment (aripiprazole) [inline pallets/subspace/src/voting.rs:189]: Isn't it possible to use pattern matching over `mode`?

**Review by aripiprazole — APPROVED** [review]
The runtime upgrade ran successfully

File: pallets/subspace/src/voting.rs (hunk 2 above)
Comment (saiintbrisson) [reply pallets/subspace/src/voting.rs:189]: yep, ill refactor it and merge afterwards

**Pull request merged**